interpreter.asm

```diff
@@ -244,98 +232,91 @@ FIND
     jsr R_TO
     jmp ZERO
 
-    +BACKLINK
-    !byte	9
-    !text	"find-name"
+    +BACKLINK "find-name", 9
 FIND_NAME ; ( caddr u -- caddr u 0 | xt 1 | xt -1 )
-    txa
-    pha
-
     lda LSB,x
     beq .find_failed
-    sta	.findlen + 1
-    sta	.findlen2 + 1
+    sta .findlen + 1
+    sta .findlen2 + 1
 
-    lda	MSB+1,x
-    sta	W2+1
-    lda	LSB+1,x
-    sta	W2
-
-    lda W2
-    bne +
-    dec W2+1
-+   dec W2
-    lda W2
-    bne +
+    lda MSB+1,x
+    sta W2+1
+    lda LSB+1,x
+    sec
+    sbc #3
+    sta W2
+    bcs +
     dec W2+1
-+   dec W2
-
-    ldx	_LATEST
-    lda	_LATEST + 1
++
+    lda _LATEST
+    sta W
+    lda _LATEST + 1
+    sta W + 1
+    ; W now contains new dictionary pointer.
+    ldy #2
+    lda (W), y ; get string length of dictionary word
 .examine_word
-    sta	W + 1
-    stx	W
-    ; W now contains new dictionary word.
-
-    ldy	#2
-    lda	(W), y ; get string length of dictionary word
-    and	#STRLEN_MASK | F_HIDDEN ; include hidden flag... so we don't find the hidden words.
+    and #STRLEN_MASK | F_HIDDEN ; include hidden flag... so we don't find the hidden words.
 .findlen
-    cmp	#0
-    beq	.string_compare
+    cmp #0
+    beq .string_compare
 
 .word_not_equal
-    ; no match, advance the linked list.
-    ldy	#0
-    lax	(W), y
-    iny
-    lda	(W), y
+    ; no match, advance the dp
```

Comment:
The other advantage is that removing the linked list allows the dictionary to be portable.  Is compile / interpret time a critical goal that shouldn't be overlooked?

Reply:
It is something to keep in mind. In the past, I have timed how long it takes to make a full build, trying to keep it not more painfully slow than needed :)
For now, I suggest no changes to this code for the sake of speed. Once everything is in place and the wordlist is stripped from useless entries, it will be very interesting to again measure the full build time.